### PR TITLE
Remove dpi scaling of maximum width

### LINKED ISSUE #1528: [Bug]: `use_xft = true` breaks window sizing
### What happened?

Window sizing is smaller when `use_xft = true` is set then when it is not, if a window is a forced size with min_width and max_width being the same then the window size will be slightly smaller then that.

This first image is with `use_xft=true` the colored temperature shape should be centered 
shot-2023-05-09_18-54-42

This second shot is showing everything working properly with it disabled
shot-2023-05-09_18-55-23

Issue is seen running a few day old version of conky from git on openSUSE Tumbleweed

### Version

git

### Which OS/distro are you seeing the problem on?

Linux (other)

### Conky config

```Lua
https://github.com/philer/polycore  `/opt/conky/bin/conky -c  conkyrc.lua` I had to comment out line 45 of layout.lua to stop a crash as I don't have gpu support.
```

### Stack trace

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Caellian**
This PR removes `dpi_scale`ing of maximum width setting.

This seems incorrect because I expect maximum width to use absolute pixel value, otherwise meaning of `maximum_width` setting changes with DPI:

image
<sub>NOTE: <em>Height difference is caused by font height, conky height can't be limited.</em></sub>

This fixes #1528.

This PR also fixes a regression introduced in #1841, where I incorrectly assumed templated base class function would be overriden by display outputs so DPI scaling was completely disabled.

# Other changes

This PR also:
- sets `workspace` dimensions from WL `output_geometry` because workspace is used for some calculations in `conky.cc`.
  - this requires more work for proper support as noted in comments, but it should make Wayland less broken.
- adds conky xdg dir and script parent directory to Lua path so that scripts don't have to update it manually.

# Testing

- [x] Use mock Xft dpi value and see whether example from #1528 works better with/without this PR.

### CODE REVIEW

File: src/conky.cc [hunk 1]
```diff
@@ -1961,6 +1962,40 @@ void load_config_file() {
...
+    l.getfield(-1, "path");
+
+    auto path = l.tostring(-1);
+    path.append(path_ext);
+    l.pop();
+    l.pushstring(path.c_str());
+
+    l.setfield(-2, "path");
+    l.pop();
+  }
```
Comment (Caellian) [inline src/conky.cc:1997]: Adding `$HOME/.config/conky` and config parent directory to lua path before loading the script.

File: src/display-x11.cc [hunk 2]
```diff
@@ -945,18 +945,14 @@ void display_output_x11::move_win(int x, int y) {
...
 
-const size_t PIXELS_PER_INCH = 96;
-template <typename T, typename>
-T display_output_x11::dpi_scale(T value) {
-#if defined(BUILD_XFT)
+const float PIXELS_PER_INCH = 96.0;
+float display_output_x11::get_dpi_scale() {
+#ifdef BUILD_XFT
   if (use_xft.get(*state) && xft_dpi > 0) {
-    return (value * xft_dpi + (value > 0 ? 48 : -48)) / PIXELS_PER_INCH;
```
Comment (Caellian) [inline src/display-x11.cc:953]: I believe `(value > 0 ? 48 : -48)` was rounding the result for `int` values before, so I replaced it with `ceil` and `floor` in new code.

File: src/display-wayland.cc [hunk 3]
```diff
@@ -300,7 +300,17 @@ static void output_geometry(void *data, struct wl_output *wl_output, int32_t x,
...
+                            int32_t transform) {
+  // TODO: Add support for proper output management through:
+  // - xdg-output-unstable-v1
+  // Maybe also support (if XDG protocol not reported):
+  // - kde-output-management(-v2)
+  // - wlr-output-management-unstable-v1
+  workarea[0] = x;  // TODO: use xdg_output.logical_position
+  workarea[1] = y;
+  workarea[2] = physical_width;
+  workarea[3] = physical_height;
```
Comment (Caellian) [inline src/display-wayland.cc:312]: This is the workarea change.

File: src/conky.cc [hunk 4]
```diff
@@ -275,7 +276,7 @@ int text_width = 1,
 struct information info;
 
 /* path to config file */
-std::string current_config;
+std::filesystem::path current_config;
```
Comment (brndnmtthws) [inline src/conky.cc:279]: Nice

**Review by brndnmtthws — APPROVED** [review]
Seems reasonable.

**Pull request merged**